### PR TITLE
fix: [CDS-78394]: move instance task details query to secondary node

### PR DESCRIPTION
**Pull request by abhinavmittal-harness**
## Describe your changes

Looking at the usage of the query being used for InstanceSyncTaskDetails.
It is safe to move it to analytics node.
As this entity is used every 10 mins where we fetch the entity, do some processing, based on the processing we wither update or delete the entity on that run in the very end.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52440)

### CODE REVIEW

File: 400-rest/src/main/java/software/wings/instancesyncv2/service/CgInstanceSyncTaskDetailsService.java [hunk 1]
```diff
@@ -84,15 +84,15 @@ public List<InstanceSyncTaskDetails> fetchAllForPerpetualTask(String accountId,
   }
 
   public boolean isInstanceSyncTaskDetailsExist(String accountId, String perpetualTaskId) {
-    Query<InstanceSyncTaskDetails> query = mongoPersistence.createQuery(InstanceSyncTaskDetails.class)
```
Comment (namantalaycha10) [inline 400-rest/src/main/java/software/wings/instancesyncv2/service/CgInstanceSyncTaskDetailsService.java:87]: @abhinavmittal-harness , Are we not doing this Analytics Query changes for all the query of instanceSyncTaskDetails for example these methods in below SS ?
image

Comment (abhinavmittal-harness) [reply 400-rest/src/main/java/software/wings/instancesyncv2/service/CgInstanceSyncTaskDetailsService.java:87]: We can but those were not flagged in the alert
If you want I can do the same for them as well
Let me raise a commit for other as well

**Review by namantalaycha10 — DISMISSED** [review]

**Comment by namantalaycha10** [comment]
@abhinavmittal-harness , please take care that these changes should be validated (whole k8s instance sync v2 CG cases) before merging as this instance sync is a very sensitive code

**Review by abosii-harness — APPROVED** [review]

**Review by abosii-harness — APPROVED** [review]

**Comment by abhinavmittal-harness** [comment]
trigger uts

**Comment by abhinavmittal-harness** [comment]
trigger messagecheck

**Comment by abhinavmittal-harness** [comment]
trigger review

**Pull request merged**